### PR TITLE
Add descriptive page titles and more

### PR DESCRIPTION
**Pull request by rachaelcodes**
## Context

Our page `<title>`s were not accessibility guidelines compliant, as we had the same title throughout the site. Similarly, not all pages had a `<h1>`. 

Not an accessibility fix, but annoying: the tags showing search choices had a misaligned cross.

## Changes proposed in this pull request

1. All page templates have a specific `<title>`
2. All page templates contain a `<h1>`
3. The cross on the `facet-tag`s has been centred:

### Before
Screenshot 2024-01-02 at 14 27 40

### After
Screenshot 2024-01-02 at 14 27 46

4. The two base templates have been merged. (The only visible change is that the FAQ link is visible on the logged-out view, which I think it should be as it's not a protected page).

## Guidance to review

1. Check the page titles in the browser
2. Check the new `<h1>` tags are paired and that the next header tag is `<h2>` etc
3. Visit the evaluation search page and try out different filter options
4. View the logged in and logged out page layouts

## Link to JIRA ticket

https://technologyprogramme.atlassian.net/browse/ER-95
https://technologyprogramme.atlassian.net/browse/ER-62

## Things to check

~- [ ] I have added any new ENV vars in all deployed environments~

### CODE REVIEW

File: evaluation_registry/templates/auth/base.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <html lang="en" class="govuk-template ">
   <head>
       <meta charset="UTF-8">
-      <title>Evaluation Registry</title>
+      <title>{% block head_title %}{% endblock %} — Evaluation Registry</title>
```
Comment (duncanjbrown) [inline evaluation_registry/templates/auth/base.html:5]: This removes "- GOV.UK from the title — is that correct?

File: evaluation_registry/templates/base.html [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <html lang="en" class="govuk-template ">
   <head>
     <meta charset="UTF-8">
-    <title>Evaluation Registry</title>
+    <title>{% block head_title %}{% endblock %} — Evaluation Registry</title>
```
Comment (duncanjbrown) [inline evaluation_registry/templates/base.html:5]: Slightly OT: it looks like we've got two very similar base templates, which means this change has to appear twice — is there a good reason not to have one template?

File: evaluation_registry/templates/faq.html [hunk 3]
```diff
@@ -1,4 +1,7 @@
 {% extends "base.html" %}
+{% block head_title %}
+  Frequently-Asked Questions
```
Comment (duncanjbrown) [inline evaluation_registry/templates/faq.html:3]: GDS style is [no hyphens](https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style#faqs-frequently-asked-questions)

File: evaluation_registry/templates/auth/base.html (hunk 1 above)
Comment (duncanjbrown) [reply evaluation_registry/templates/auth/base.html:5]: 🚨 extreme nitpick: the [service manual](https://www.gov.uk/service-manual/design/writing-for-user-interfaces#style) prefers dashes `-` to em-dashes `—`

**Review by duncanjbrown — COMMENTED** [review]
Looks good! Minor comments inline — I think it'd be good to dedupe those templates if possible?

I'm just figuring out how to get the CSS changes to show up on my local copy.

**Comment by duncanjbrown** [comment]
CSS looks good 👍

File: evaluation_registry/templates/auth/base.html (hunk 1 above)
Comment (rachaelcodes) [reply evaluation_registry/templates/auth/base.html:5]: > This removes "- GOV.UK from the title — is that correct?
I *think* so, as we're hosting on *.cabinetoffice.gov.uk not *.service.gov.uk (which also means we should update our header etc - there is a ticket for this), but am happy to be corrected.

File: evaluation_registry/templates/base.html (hunk 2 above)
Comment (rachaelcodes) [reply evaluation_registry/templates/base.html:5]: Hm good point, will take a look

File: evaluation_registry/templates/auth/base.html (hunk 1 above)
Comment (duncanjbrown) [reply evaluation_registry/templates/auth/base.html:5]: that sounds reasonable!

**Review by duncanjbrown — APPROVED** [review]
LGTM!

**Pull request merged**